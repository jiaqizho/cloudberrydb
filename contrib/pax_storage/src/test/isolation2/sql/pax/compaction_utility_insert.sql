-- @Description Tests the compaction of data inserted in utility mode
-- 
DROP TABLE IF EXISTS foo;
CREATE TABLE foo (a INT, b INT, c CHAR(128)) distributed randomly;
CREATE INDEX foo_index ON foo(b);

0U: INSERT INTO foo VALUES (2, 2, 'c');
0U: INSERT INTO foo VALUES (3, 3, 'c');
SELECT segment_id, pttupcount FROM get_pax_aux_table_all('foo');
DELETE FROM foo WHERE a = 2;
UPDATE foo SET b = -1 WHERE a = 3;
VACUUM foo;
SELECT segment_id, pttupcount FROM get_pax_aux_table_all('foo') where pttupcount > 0;
