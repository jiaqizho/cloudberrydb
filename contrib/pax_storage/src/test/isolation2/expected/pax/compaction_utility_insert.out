-- @Description Tests the compaction of data inserted in utility mode
--
DROP TABLE IF EXISTS foo;
DROP
CREATE TABLE foo (a INT, b INT, c CHAR(128)) distributed randomly;
CREATE
CREATE INDEX foo_index ON foo(b);
CREATE

0U: INSERT INTO foo VALUES (2, 2, 'c');
INSERT 1
0U: INSERT INTO foo VALUES (3, 3, 'c');
INSERT 1
SELECT segment_id, pttupcount FROM get_pax_aux_table_all('foo');
 segment_id | pttupcount 
------------+------------
 0          | 1          
 0          | 1          
(2 rows)
DELETE FROM foo WHERE a = 2;
DELETE 1
UPDATE foo SET b = -1 WHERE a = 3;
UPDATE 1
VACUUM foo;
VACUUM
SELECT segment_id, pttupcount FROM get_pax_aux_table_all('foo') where pttupcount > 0;
 segment_id | pttupcount 
------------+------------
 0          | 1          
 0          | 1          
 0          | 1          
(3 rows)
